plugins.go

```diff
@@ -0,0 +1,78 @@
+package goflow
+
+import (
+	"fmt"
+	"io/ioutil"
+	"path"
+	"plugin"
+	"strings"
+)
+
+const plugInSuffix = `_goplug.so`
+
+//PlugIn something
+type PlugIn interface {
```

Comment:
Are `params` mandatory for all plugins? Can't we just load _any_ components as plugins? Forcing all plugin components adhere to this interface is quite restrictive.

Also, I have a strong feeling that `Parameters` is reinventing IIPs. In FBP, IIPs (initial IPs) are used to send pre-defined input to components. See _graph_iip_test.go_ for example.